notebooks/notebook_support.py

```diff
@@ -127,10 +274,25 @@ def load(self) -> Tuple["NDArray[np.float_]", "NDArray[np.float_]"]:
         self.model.load_state_dict(
             torch.load(self.data_dir / f"{self.name}_weights.pth")
         )
+        self.model.eval()
         with open(self.data_dir / f"{self.name}_train_val_loss.pkl", "rb") as file:
             return pkl.load(file)
 
 
+def process_imgnet_io(
+    df: pd.DataFrame, labels: dict
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    x = df["normalized_images"]
+    y = df["labels"]
+    ds_label_to_model_label = {
+        ds_label: idx for idx, ds_label in enumerate(labels.values())
+    }
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+    x_nn = torch.stack(x.tolist()).to(device)
+    y_nn = torch.tensor([ds_label_to_model_label[yi] for yi in y], device=device)
+    return x_nn, y_nn
+
+
 def plot_dataset(
     train_ds: Tuple["NDArray[np.float_]", "NDArray[np.int_]"],
```

Comment:
Can't we use our `Dataset`? (asking)

Reply:
The Dataset class is not optimal for torch datasets at the moment. Main issues:
- it does not have validation set
- it runs checks with sklearn which would need to convert torch tensors, with memory overhead
- Apart for wrapping train and test sets, it does not add anything more since its methods are not used in influences (e.g. subsampling of indices)
- Not compatible with torch DataLoaders